backend/src/main/java/gov/cdc/usds/simplereport/service/ResultsUploaderDeviceValidationService.java

```diff
@@ -0,0 +1,58 @@
+package gov.cdc.usds.simplereport.service;
+
+import static gov.cdc.usds.simplereport.config.CachingConfig.DEVICE_MODEL_AND_TEST_PERFORMED_CODE_SET;
+
+import gov.cdc.usds.simplereport.db.repository.DeviceTypeRepository;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.cache.annotation.Caching;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ResultsUploaderDeviceValidationService {
+  private final DeviceTypeRepository deviceTypeRepository;
+
+  @Cacheable(DEVICE_MODEL_AND_TEST_PERFORMED_CODE_SET)
```

Comment:
Cool stuff here! I'll have a little reading to do on some of the implementation details but this is a nice addition.